plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -205,90 +200,53 @@ public void setProject(IProject project) {
 	}
 
 	@Override
-	public String getName() {
-		return this.name;
+	public String getResourceName() {
+		return this.resourceName;
 	}
 
 	@Override
-	public void setName(String name) {
-		firePropertyChange(PROPERTY_NAME, this.name, this.name = name);
+	public void setResourceName(String resourceName) {
+		firePropertyChange(PROPERTY_RESOURCE_NAME, this.resourceName, this.resourceName = resourceName);
 	}
 
 	@Override
-	public String getImage() {
-		return this.image;
+	public String getImageName() {
+		return this.imageName;
 	}
 
 	@Override
-	public void setImage(String image) {
-		if(StringUtils.isBlank(image)) return;
-		firePropertyChange(PROPERTY_IMAGE, this.image, this.image = image);
-		DockerImageURI uri = new DockerImageURI(image);
-		setName(uri.getName());
-		resetImageInfo();
-		initContainerInfo();
+	public void setImageName(final String imageName) {
+		if(StringUtils.isBlank(imageName)) {
+			return;
+		}
+		firePropertyChange(PROPERTY_IMAGE_NAME, this.imageName, this.imageName = imageName);
+		final DockerImageURI uri = new DockerImageURI(imageName);
+		setResourceName(uri.getName());
 	}
 	
-	private void resetImageInfo() {
-		setEnvironmentVariables(new ArrayList<>());
-		setVolumes(new ArrayList<>());
-		setPortSpecs(new ArrayList<>());
-		setReplicas(DEFAULT_REPLICA_COUNT);
-		imageInfoInitialized = false;
-	}
-
 	@Override
-	public List<EnvironmentVariable> getEnvironmentVariables() {
-		return environmentVariables;
-	}
-	
-	private synchronized void initContainerInfo() {
-		IDockerImageMetadata info = getDockerMetaData();
-		if(info == null) return;
-		setEnvVars(info.env());
-		
-		List<IPort> portSpecs = new ArrayList<>();
-		Set<String> specs = info.exposedPorts();
-		for (String spec : specs) {
-			try {
-				portSpecs.add(new PortSpecAdapter(spec));
-			}catch(IllegalArgumentException e) {
-				OpenShiftUIActivator.getDefault().getLogger().logError(e);
-			}
+	public boolean initializeContainerInfo() {
+		this.imageMeta = lookupImageMetadata();
+		if (this.imageMeta == null) {
+			return false;
 		}
+		final List<EnvironmentVariable> envVars = this.imageMeta.env().stream().map(env -> env.split("="))
```

Comment:
@xcoulon , good that my comment helped you to find one more case to consider, but my idea was not that '=' is absent but that name=value has empty value so that literally string is 'name='. In this case split("=") returns array of length 1. 

If this case can never happen, then there is nothing to do for it, but it is good that an incorrect question may help bring to light a real problem.